internal/controller/validator.go

```diff
@@ -173,6 +173,26 @@ func (r *DataProtectionApplicationReconciler) ValidateDataProtectionCR(log logr.
 
 		}
 
+		enforcedRestoreSpec := r.dpa.Spec.NonAdmin.EnforceRestoreSpec
+
+		if enforcedRestoreSpec != nil {
+			if len(enforcedRestoreSpec.ScheduleName) > 0 {
+				return false, fmt.Errorf(NACNonEnforceableErr, "spec.nonAdmin.enforcedRestoreSpec.scheduleName")
+			}
+
+			if enforcedRestoreSpec.IncludedNamespaces != nil {
+				return false, fmt.Errorf(NACNonEnforceableErr, "spec.nonAdmin.enforcedRestoreSpec.includedNamespaces")
+			}
+
+			if enforcedRestoreSpec.ExcludedNamespaces != nil {
```

Comment:
Can't this become a feature, where an admin want some namespaces to be restricted and not allow users from those namespaces to backup or restore workload? I know it's not part of this PR but as a future request, which will need to change also backup enforcedBackupSpec.ExcludedNamespaces part ?

Reply:
I think you talking along the lines where or which NS we want NAC B/R to work ?

Reply:
Yes

Reply:
Yeah so thats a separate feature where we want to allow admins to configure NAC for a set of namespaces based on some list or criteria. I can track it in a issue.

Reply:
Yeah -- restricting namespace would mean "we shouldn't have NAB/NAR at all", not "namespace field in NAB/NAR adds restrictions".

Reply:
@shubham-pampattiwar Hmm. So couldn't we handle the "set of namespaces to allow NAC for" via rbac? i.e. don't grant create NAB/NAR role for users on namespaces where we don't want to enable NAC?

Reply:
Agree, no need to change anything, just thinking laud here.